buildenv/jenkins/JenkinsfileBase

```diff
@@ -981,17 +982,40 @@ def addGrinderLink() {
 def addFailedTestsGrinderLink(){
 	if (currentBuild.result == 'UNSTABLE' || currentBuild.result == 'FAILURE' || currentBuild.result == 'ABORTED') {
 		echo "Fetching Tap files..."
-		def tapFiles = findFiles(glob: "**/*.tap")
 		def failedTestList = "testList+TESTLIST="
-		for (def tapFile: tapFiles) {
-			echo "Tap file found: ${tapFile}"
-			def file = readFile(file: "${tapFile}")
-			TapConsumer tapConsumer = TapConsumerFactory.makeTap13Consumer()
-			TestSet testSet = tapConsumer.load(file)
-			for (int i = 1; i <= testSet.getNumberOfTestResults(); i++) {
-				if (!testSet.getTestResult(i).getStatus().toString().equals("ok")) {
-					def testCase = testSet.getTestResult(i).getDescription().substring(2)
-					failedTestList += "$testCase,"
+		if ("".equals(env.FAILED_PARALLEL_TEST_PATH)) {
+			def buildId = currentBuild.id
+			def name = currentBuild.projectName
+			def tapFiles = findFiles(glob: "${name}/${buildId}/**/*.tap")
+			for (def tapFile: tapFiles) {
+				echo "Tap file found: ${tapFile}"
+				def file = readFile(file: "${tapFile}")
+				TapConsumer tapConsumer = TapConsumerFactory.makeTap13Consumer()
+				TestSet testSet = tapConsumer.load(file)
+				echo "Testset ${testSet}"
+				for (int i = 1; i <= testSet.getNumberOfTestResults(); i++) {
+					if (!testSet.getTestResult(i).getStatus().toString().equals("ok")) {
+						def testCase = testSet.getTestResult(i).getDescription().substring(2)
+						failedTestList += "$testCase,"
+					}
+				}
+			}
+		}
+		else {
+			def buildPaths = env.FAILED_PARALLEL_TEST_PATH.substring(0, env.FAILED_PARALLEL_TEST_PATH.length() - 1).split(',')
+			for (def buildPath: buildPaths) {
+				def tapFiles = findFiles(glob: "${buildPath}/**/*.tap")
+				for (def tapFile: tapFiles) {
+					echo "Tap file found: ${tapFile}"
+					def file = readFile(file: "${tapFile}")
+					TapConsumer tapConsumer = TapConsumerFactory.makeTap13Consumer()
+					TestSet testSet = tapConsumer.load(file)
+					for (int i = 1; i <= testSet.getNumberOfTestResults(); i++) {
+						if (!testSet.getTestResult(i).getStatus().toString().equals("ok")) {
+							def testCase = testSet.getTestResult(i).getDescription().substring(2)
+							failedTestList += "$testCase,"
+						}
+					}
```

Comment:
- The majority of code in if-else is duplicated. We just need else case and define `buildPaths` accordingly.
- We do not need the Jenkins environment variable - env.FAILED_PARALLEL_TEST_PATH, a local variable should be passed into the function.